crates/matrix-sdk-ui/src/timeline/pinned_events_loader.rs

```diff
@@ -180,7 +186,28 @@ impl PinnedEventsRoom for Room {
             }
 
             debug!("Loading pinned event {event_id} from HS");
-            self.event(event_id, request_config).await.map(|e| (e, Vec::new()))
+            let event = self.event(event_id, request_config).await?;
+
+            debug!("Loading relations of pinned event {event_id} from HS");
+            let mut related_events = Vec::new();
+            let mut opts = RelationsOptions {
+                include_relations: IncludeRelations::AllRelations,
+                recurse: true,
+                limit: Some(uint!(1000)),
+                ..Default::default()
+            };
+
+            loop {
+                let relations = self.relations(event_id.to_owned(), opts.clone()).await?;
+                related_events.extend(relations.chunk);
+                if let Some(next_from) = relations.next_batch_token {
+                    opts.from = Some(next_from);
+                } else {
+                    break;
+                }
+            }
```

Comment:
One more question. We think it would be good to store these events in the cache after downloading them. We think something like this might work:

```
if let Ok((cache, _handles)) = self.event_cache().await {
            cache.save_events(related_events).await;
        }
```

If that seems like too much of a rabbit-hole let me know and we can make it separate.